model/dependencies.go

```diff
@@ -6,33 +6,37 @@ import (
 )
 
 type dependencyIncluder struct {
-	Project   *Project
-	requester string
-	included  map[TVPair]bool
+	Project                *Project
+	requester              string
+	included               map[TVPair]bool
+	deactivateGeneratedDep map[TVPair]bool
 }
 
 // IncludeDependencies takes a project and a slice of variant/task pairs names
 // and returns the expanded set of variant/task pairs to include all the dependencies/requirements
 // for the given set of tasks.
 // If any dependency is cross-variant, it will include the variant and task for that dependency.
 // This function can return an error, but it should be treated as an informational warning
-func IncludeDependencies(project *Project, tvpairs []TVPair, requester string) ([]TVPair, error) {
+// activationInfo and generatedVariants are required in the case for generate tasks to detect if
+// new generated dependency's task/variant pairs are depended on by inactive tasks. If so,
+// we also set these new dependencies to inactive.
+func IncludeDependencies(project *Project, tvpairs []TVPair, requester string, activationInfo *specificActivationInfo, generatedVariants []parserBV) ([]TVPair, error) {
```

Comment:
Organizational nit: can we have two functions IncludeDependencies and IncludeDependenciesWithGenerated (or etc)?
IncludeDependenciesWithGenerated can have this function signature ^^^ and IncludeDependencies can just wrap IncludeDependencesWithGenerated but pass in nil, nil for the last two arguments. That would clean up the usage of this in everywhere but generated tasks.